lib/api/core/plugins/pluginsManager.js

```diff
@@ -749,36 +902,61 @@ class PluginsManager {
 
         Object.keys(route).forEach(key => {
           if (routeProperties.indexOf(key) === -1) {
-            throw new PluginImplementationError(
-              `${errorRoutePrefix} Unknown property "${key}" in route definition.`
-              + didYouMean(key, routeProperties));
+            errorsManager.throw(
+              'internal',
+              'plugins',
+              'unknown_property_key_in_route_definition',
+              errorRoutePrefix,
+              key,
+              didYouMean(key, routeProperties)
+            );
           }
 
           if (typeof route[key] !== 'string'
             || (route[key].length === 0 && key !== 'url'))
           {
-            throw new PluginImplementationError(
-              `${errorRoutePrefix} "${key}" must be a non-empty string`);
+            errorsManager.throw(
+              'internal',
+              'plugins',
+              'key_cannot_be_empty_string',
+              errorRoutePrefix,
+              key
+            );
           }
         });
 
         if (!this.controllers[controllerName]) {
-          throw new PluginImplementationError(
-            `${errorRoutePrefix} Undefined controller "${route.controller}".`
-            + didYouMean(route.controller, controllerNames));
+          errorsManager.throw(
+            'internal',
+            'plugins',
+            'undefined_controller',
+            errorRoutePrefix,
+            route.controller,
+            didYouMean(route.controller, controllerNames)
+          );
         }
 
         if (!this.controllers[controllerName][route.action]) {
           const actionNames = Object.keys(this.controllers[controllerName]);
-          throw new PluginImplementationError(
-            `${errorRoutePrefix} Undefined action "${route.action}".`
-            + didYouMean(route.action, actionNames));
+          errorsManager.throw(
+            'internal',
+            'plugins',
+            'undefined_action',
+            errorRoutePrefix,
+            route.action,
+            didYouMean(route.action, actionNames)
+          );
         }
 
         if (httpVerbs.indexOf(route.verb.toLowerCase()) === -1) {
-          throw new PluginImplementationError(
-            `${errorRoutePrefix} Only following http verbs are allowed: "${httpVerbs.join(', ')}".`
-            + didYouMean(route.verb, httpVerbs));
+          errorsManager.throw(
+            'internal',
+            'plugins',
+            'http_verb_not_allowed',
+            errorRoutePrefix,
+            httpVerbs.join(', '),
```

Comment:
You don't have to join, this is redundant with what [Array.toString](https://devdocs.io/javascript/global_objects/array/tostring) does.

Reply:
Not exactly, `Array.toString` does not add a whitespace